includes/Core/REST_API/REST_Routes.php

```diff
@@ -281,6 +343,59 @@ function ( $response ) {
 					),
 				)
 			),
+			new REST_Route(
+				'core/user/data/user-input-settings',
+				array(
+					array(
+						'permission_callback' => '__return_true', // It gets checked in the User_Input_Settings class.
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$user_input_settings = new User_Input_Settings( $this->context, $this->authentication );
+							return rest_ensure_response( $user_input_settings->get_settings() );
+						},
+					),
+					array(
+						'permission_callback' => '__return_true', // It gets checked in the User_Input_Settings class.
+						'methods'             => WP_REST_Server::CREATABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$user_input_settings = new User_Input_Settings( $this->context, $this->authentication );
+							return rest_ensure_response(
+								$user_input_settings->set_settings(
+									$request->get_param( 'settings' )
+								)
+							);
+						},
+						'args'                => array(
+							'settings' => array(
+								'type'       => 'object',
+								'required'   => true,
+								'properties' => array(
```

Comment:
Unfortunately I don't think we can define the `args` in this way since the incoming post fields are actually nested under a top-level `data` key.

Reply:
Ok, updated it to be nested under `data` object.